js/ui/viewSelector.js

```diff
@@ -445,6 +449,37 @@ const ViewsClone = new Lang.Class({
                                    transition: 'easeOutQuad' });
             }
         }));
+
+        this._settings.connect('changed::' + KEY_TEXT_SCALING_FACTOR, Lang.bind(this, function() {
+            /*
+             * Add it in an idle to give Clutter some time to
+             * recalculate whatetever it needs to. Simply queueing
+             * redraw right after the setting changes doesn't
+             * work.
```

Comment:
Sorry for the back and forth on this, but I believe I understand the reason you can't just queue a redraw right away here.
Clutter does not listen to this gsetting directly; however it listens to the `Xft/DPI` and `Gdk/UnscaledDPI` xsettings, which gnome-settings-daemon is responsible of updating whenever the gsetting changes.

In other words, you should be able to connect to notifications on the `font-dpi` property of `ClutterSettings` and avoid the idle callback.